sdk/identity/azure-identity/src/main/java/com/azure/identity/AadCredentialBuilderBase.java

```diff
@@ -107,4 +107,16 @@ public T additionallyAllowedTenants(List<String> additionallyAllowedTenants) {
         identityClientOptions.setAdditionallyAllowedTenants(IdentityUtil.resolveAdditionalTenants(additionallyAllowedTenants));
         return (T) this;
     }
+
+    /**
+     * Disables instance discovery.
+     *
+     * @return An updated instance of this builder with instance discovery disabled.
+     */
+    @SuppressWarnings("unchecked")
+
+    public T disableInstanceDiscovery() {
```

Comment:
Our argument name is instance_discovery which defaults to None (use whatever msal has)

If user wants to opt out, they pass in instance_discovery=False.

Reply:
Understood. This is how we've represented similar concepts in Java. See also [`DeviceCodeCredentialBuilder.disableAutomaticAuthentication`](https://github.com/Azure/azure-sdk-for-java/blob/ae606de127c9aeeedd7ee8e0d68bf141f32e3ff9/sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredentialBuilder.java#L74).